plugin/pkg/shoot/validator/admission.go

```diff
@@ -831,34 +838,65 @@ func validateAlicloud(c *validationContext) field.ErrorList {
 	return allErrs
 }
 
-func validateDNSConstraints(constraints []garden.DNSProviderConstraint, provider, oldProvider *string) (bool, []string) {
-	if apiequality.Semantic.DeepEqual(provider, oldProvider) {
-		return true, nil
+func validateProvider(c *validationContext) field.ErrorList {
+	var (
+		allErrs = field.ErrorList{}
+		path    = field.NewPath("spec", "provider")
+	)
+
+	if c.seed != nil {
+		allErrs = append(allErrs, admissionutils.ValidateNetworkDisjointedness(c.seed.Spec.Networks, garden.K8SNetworks{Nodes: &c.shoot.Spec.Networking.Nodes, Pods: c.shoot.Spec.Networking.Pods, Services: c.shoot.Spec.Networking.Services}, path.Child("networking"))...)
```

Comment:
Sorry, wrong location. This is the place causing the admission error. ` admissionutils.ValidateNetworkDisjointedness` returns two errors - `spec.provider.networking.pods required` and `spec.provider.networking.services required`. With the master branch I am able to apply the same Shoot spec.

Reply:
Hm, can you share your `Shoot` and `Seed` resources?
If I remember correctly then this validation was missing anyways, but I'm not 100% sure anymore.

Reply:
```yaml
apiVersion: garden.sapcloud.io/v1beta1
kind: Seed
metadata:
  # [...]
spec:
  networks:
    nodes: 10.250.0.0/16
    pods: 100.96.0.0/11
    services: 100.64.0.0/13
    shootDefaults: {}
---
apiVersion: garden.sapcloud.io/v1beta1
kind: Shoot
metadata:
  # [...]
spec:
  cloud:
    aws:
      networks:
        internal:
        - 10.222.112.0/22
        nodes: 10.222.0.0/16
        pods: 10.223.128.0/17
        public:
        - 10.222.96.0/22
        services: 10.223.0.0/17
        vpc:
          cidr: 10.222.0.0/16
        workers:
        - 10.222.0.0/19

```

Reply:
Ahh, okay, nice catch!!! What about now?

Reply:
Now is fixed.